pandas/core/indexes/base.py

```diff
@@ -359,6 +360,9 @@ def _outer_indexer(
     _is_numeric_dtype: bool = False
     _can_hold_na: bool = True
     _can_hold_strings: bool = True
+    # Whether this index is a NumericIndex, but not a Int64Index, Float64Index,
+    # UInt64Index or RangeIndex
```

Comment:
this is a very confusing flag consider we already have is_numeric_dtype

if it's really necessary then call it something else

Reply:
actually prefer a function to do this is think (just a free function)

you need this for testing anyhow

Reply:
`_is_numeric_dtype` is True for all numeric indexes (NumericIndex, Int64Index, RangeIndex etc.) This should be kept after pandas 2.0.

`_is_numeric_index` is True for NumericIndex, but False for Int64Index, RangeIndex etc. This should be removed for pandas 2.0 and is just a compatability flag, because we can't do `isinstance(idx, NumericIndex`.